### ISSUE TITLE
pigeon classify no output

### ISSUE THREAD

**Issue by sum732** [PacificBiosciences/pigeon#2]
Hello,

Thanks for creating Pigeon with most SQANTI3 in it.

I am using piegon for my bulk RNA-Seq data.

I am following the workflow (https://isoseq.how/classification/workflow) and at things are stuck at classification, its been >13 hrs and still no output. 
```
pigeon classify ../collapse/0456_collapsed.sorted.gff /data/chm13.draft_v2.0.gene_annotation_v2.2.gtf /data/Reference/chm13v2.0.fa --fl ../collapse/0456_collapsed.abundance.txt -j 12
```
```
pigeon --version
pigeon 1.0.0 (commit -v1.0.0)

Using:
  pbbam     : 2.2.0 (commit v2.2.0-1-g8c081f6)
  pbcopper  : 2.1.0 (commit v2.1.0)
  boost     : 1.77
  htslib    : 1.14
  zlib      : 1.2.11
```
Can you please help me with this. How to proceed. 

BW,
SM

**Comment by derekwbarnett** [comment]
Hi @sum732,
Please `pigeon sort` your genome annotation file prior to indexing, and then use the sorted data as input to `pigeon classify`:

```
pigeon sort chm13.draft_v2.0.gene_annotation_v2.2.gtf -o chm13.draft_v2.0.gene_annotation_v2.2.sorted.gtf
pigeon index chm13.draft_v2.0.gene_annotation_v2.2.sorted.gtf
pigeon classify ... chm13.draft_v2.0.gene_annotation_v2.2.sorted.gtf ...
```

Apologies for the oversight in the documentation. We are updating that currently. 

The hanging behavior, when encountering similar reference file issues, has been fixed internally. It will be part of the next bioconda update. But adding the `pigeon sort` step should hopefully get you back on track immediately.

**Comment by sum732** [comment]
Hi @derekwbarnett,
Thanks for a quick response, really much appreciated. 

I did try to sort using pigeon, but I am shown following error message:
```
| 20230222 17:29:32.888 | FATAL | pigeon sort ERROR: basic_string::_M_replace
```
I used IGV to sort and used that gave me 
```
 chm13.draft_v2.0.gene_annotation_v2.2.sorted.gtf
```
pigeon index 
```
chm13.draft_v2.0.gene_annotation_v2.2.sorted.gtf.pgi
```
to create the index and tried to classify. But still no movement. 

```
pigeon classify ../collapse/0456_collapsed.sorted.gff /data/chm13.draft_v2.0.gene_annotation_v2.2.sorted.gtf/data/Reference/chm13v2.0.fa --log-file classify_local.log --log-level INFO -o local_classification_0456 -j 4 
```

**Comment by derekwbarnett** [comment]
After `pigeon sort` exits with an error, do you have any leftover files that look something like `<output_filename>.<ref_name>`? Trying to get an idea of how far it got before hitting that.

**Comment by sum732** [comment]
No errors, just gives the error above and exit.

**Comment by sum732** [comment]
Hi @derekwbarnett,
I downloaded the GTF again and was able to work with it. I will close this one and open another one that I discovered, not related to this one. 
Thanks again!

**Comment by derekwbarnett** [comment]
Glad you got it working.

**Comment by sum732** [comment]
Thanks @derekwbarnett , it worked with a version of GTF file and not "the" version of GTF that I need. 
I am working with CHM13hTERT reference and they have provided a GFF3 file. 

Pigeon wants GTF, so i converted one and tried that, it gave the error (as above). To test, I tried another flavor of GTF and that worked.

I have requested at T2T if they can also provide a GTF file. Meanwhile, If i share the GTF file I have, can you please see why Pigeon is not working on that? Would be of great help. 

Best Regards,
SM

**Comment by derekwbarnett** [comment]
Interesting. Yeah, if you can point me to those files, I'd be happy to check it out. I can't promise any specific time frame as I would need to budget some time to dive in. But we definitely want to keep things as seamless for users as we can.

**Comment by sum732** [comment]
Thanks so much. I understand, and hope you are able to find a solution soon. My vested interest is as I am to present this data soon.  
The original GFF3 is [here](https://s3-us-west-2.amazonaws.com/human-pangenomics/T2T/CHM13/assemblies/annotation/chm13.draft_v2.0.gene_annotation.gff3)
Converted to GTF via `agat` is [here](https://www.dropbox.com/s/teumd7fiv7d6rn5/chm13.draft_v2.0.gene_annotation_v2.2.agat_MOD.gtf?dl=1)

**Comment by grendon** [comment]
I am having difficulties with pigeon sort to prepare my reference genome. 

The version of pigeon I am using is the one that comes with smrtlink v 11.1

The genome files were downloaded from the NCBI RefSeq ftp site:
https://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_other/Anas_platyrhynchos/latest_assembly_versions/GCF_015476345.1_ZJU1.0/

The command I ran:

<pre>
pigeon sort --log-level DEBUG --log-file pigeon_sort.log  -o  ref.sorted.gtf    ref.gtf 
</pre>

pigeon did not like the header lines of that GTF file  because they started with #!.  So I changed them to ##.
Then I reran the pigeon sort command and now  aborts for a different reason. 
It sorts the first chromosome and then it quits with this error message

<pre>
$ ls
pigeon_sort.log
ref.sorted.gtf.NC_051772.1
ref.gtf

$ cat pigeon_sort.log
| 20230301 21:00:55.640 | FATAL | pigeon sort ERROR: bad optional access
</pre>

Note: the chromosome names of my ref.gft have this kind of format "NC_051772.1"  rather than the format used by gencode which are chr1 chr2 and so on.   Could that be the problem?

I would really appreciate your help.

**Comment by jmattick** [comment]
Hi @sum732 @ryansohny. Please check out our new documentation about annotations here:
https://isoseq.how/classification/pigeon-annotation.html
Additionally, we just released `pigeon v1.2.0` which has a new tool `pigeon prepare` that will validate, sort, and index your annotations. 
https://isoseq.how/classification/workflow.html#prepare-reference-files-for-pigeon
If you still encounter issues please file a new issue on our pbbioconda page
https://github.com/PacificBiosciences/pbbioconda/issues
